scripts/dts/gen_defines.py

```diff
@@ -609,15 +609,16 @@ def write_vanilla_props(node: edtlib.Node) -> None:
             if spec.enum_tokenizable:
                 as_token = prop.val_as_token
 
-                # DT_N_<node-id>_P_<prop-id>_ENUM_VAL_<val>_EXISTS 1
-                macro2val[macro + f"_ENUM_VAL_{as_token}_EXISTS"] = 1
+                if "array" not in prop.type:
```

Comment:
see https://github.com/zephyrproject-rtos/zephyr/pull/71497

Reply:
Now that #71497 is merged, all I had to do is to rebase and drop my first commit ("devicetree: Allow using enum with string-array properties").